### PR TITLE
add unit tests

### PR DESCRIPTION
**Pull request by PeterTheOne**
This PR builds on PR https://github.com/SemanticMediaWiki/SemanticCompoundQueries/pull/4.

`SCQ_CompoundQueryApiTest.php` is inspired by https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/tests/phpunit/Unit/MediaWiki/Api/AskTest.php.

The third query in `sampleQueryProvider` still fails.

### CODE REVIEW

File: tests/phpunit/SCQ_CompoundQueryApiTest.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	public function testCanConstruct() {
+		$instance = new SCQCompoundQueryApi(
+			$this->apiFactory->newApiMain( array( 'query' => 'Foo' ) ),
+			'compoundquery'
+		);
+		$this->assertInstanceOf(
+			'SCQCompoundQueryApi',
+			$instance
+		);
+	}
```
Comment (JeroenDeDauw) [inline tests/phpunit/SCQ_CompoundQueryApiTest.php]: Not sure how useful this is

File: tests/phpunit/SCQ_CompoundQueryApiTest.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+/**
+ * @covers SCQCompoundQueryApi
+ * @ingroup SemanticCompoundQueries
+ *
+ * @author Peter Grassberger < petertheone@gmail.com >
+ */
+
+class SCQCompoundQueryApiTest extends \PHPUnit_Framework_TestCase
+{
```
Comment (JeroenDeDauw) [inline tests/phpunit/SCQ_CompoundQueryApiTest.php]: misplaced bracket!

File: tests/phpunit/SCQ_CompoundQueryApiTest.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+		);
+		$this->assertInstanceOf(
+			'SCQCompoundQueryApi',
+			$instance
+		);
+	}
+
+	/**
+	 * @param array $query
+	 * @param array $expected
```
Comment (JeroenDeDauw) [inline tests/phpunit/SCQ_CompoundQueryApiTest.php]: These @param docs don't add value in such a case IMO

File: tests/phpunit/SCQ_CompoundQueryApiTest.php [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+		// If their is no printrequests array we expect an error array
+		if ( isset( $results['query']['printrequests'] ) ) {
+			$this->assertEquals( $expected, $results['query']['printrequests'] );
+		} else {
+			$this->assertArrayHasKey( 'error', $results );
+		}
+	}
+
+	public function sampleQueryProvider() {
+		// #0 Standard query
```
Comment (JeroenDeDauw) [inline tests/phpunit/SCQ_CompoundQueryApiTest.php]: It's hard to figure out what is actually tested. Apparently "standard query", and then two things which apparently are not standard. I'm not a fan of data provider usage like this (you would not need the if-else in `testExecute` if you did it without), though if you go down that path, you can add descriptive messages as keys for the argument arrays:

$provider['Foo and bar should result in baz'] = array( /* ... */ );

**Comment by JeroenDeDauw** [comment]
Really cool you are adding tests for this! :)

**Comment by PeterTheOne** [comment]
The third query in `sampleQueryProvider` still fails.

@JeroenDeDauw can you create a travis project for this repo?

**Comment by JeroenDeDauw** [comment]
> @JeroenDeDauw can you create a travis project for this repo?

Done

Good you pointed that out, without the red X I was under the impression all tests where passing :)

**Comment by PeterTheOne** [comment]
> The SemanticMediaWiki test autoloader is not available

hmm

**Comment by PeterTheOne** [comment]
Still

> The SemanticMediaWiki test autoloader is not available

**Comment by JeroenDeDauw** [comment]
Ping @mwjames

**Comment by mwjames** [comment]
> Ping @mwjames

I'd say push it (updating the fork is a pain) and once I have time I'll look into it.

**Comment by JeroenDeDauw** [comment]
> updating the fork is a pain

What do you mean? "the fork"?

**Comment by mwjames** [comment]
> What do you mean? "the fork"?

`PeterTheOne:tests` is a fork not not a branch therefore updating such of that fork is a pain as you would need to:

```
git checkout -b PeterTheOne-tests master
git pull https://github.com/PeterTheOne/SemanticCompoundQueries.git tests
```

**Comment by mwjames** [comment]
See my comments in #10.

**Comment by PeterTheOne** [comment]
@mwjames Thanks!

**Pull request merged**